build-system/server.js

```diff
@@ -79,7 +79,28 @@ app.use('/api/echo/post', function(req, res) {
   res.end(JSON.stringify(req.body, null, 2));
 });
 
+const ORIGIN_REGEX = new RegExp("http://localhost:8000|" +
```

Comment:
Also, test for the beginning of the string, since now the pattern can appear in the middle.